File: problems/2148-count-elements-with-strictly-smaller-and-greater-elements/solution.java
class Solution {
    public int countElements(int[] nums) {
        if (nums == null || nums.length <= 2) {
            return 0;
        }

        int minVal = nums[0];
        int maxVal = nums[0];

        for (int num : nums) {
            minVal = Math.min(minVal, num);
            maxVal = Math.max(maxVal, num);
        }

        int count = 0;
        for (int num : nums) {
            if (num > minVal && num < maxVal) {
                count++;
            }
        }

        return count;
    }
}

File: problems/2148-count-elements-with-strictly-smaller-and-greater-elements/solution.py
class Solution:
    def countElements(self, nums: List[int]) -> int:
        if len(nums) < 3:
            return 0
        min_val = min(nums)
        max_val = max(nums)
        count = 0
        for num in nums:
            if min_val < num < max_val:
                count += 1
        return count